### DISCUSSION TITLE
Portal not accessible after agent and portal rebrand

### DISCUSSION THREAD

**Discussion by jbob007** [Ylianst/MeshCentral#6639] [Issues] [answered]
I'm relatively new to MeshCentral—I installed it less than a week ago—and I'm still learning the ins and outs. This could very well be a user error, but I've run into an issue I haven't been able to resolve on my own, and I haven't found anyone else with the exact same problem.

Here's a breakdown of what happened:

-     Initial Setup: MeshCentral was running fine on my custom domain without any issues.
-     Customizations: I decided to customize my config file to enable agent customization and portal rebranding.
-     Issue: After making these changes, I lost access to the MeshCentral portal. Now, I receive a "Connection Refused" error, and I can't access it using either the local IP or the custom domain.

When I check the status of the MeshCentral service using `systemctl status meshcentral`, I get the following output:

```
systemctl status meshcentral
● meshcentral.service - MeshCentral Server
     Loaded: loaded (/etc/systemd/system/meshcentral.service; enabled; preset: enabled)
     Active: activating (auto-restart) since Mon 2024-12-30 08:56:49 EST; 4s ago
     Process: 885 ExecStart=/usr/bin/node /opt/meshcentral/node_modules/meshcentral/meshcentral (code=exited, status=0/SUCCESS)
     Main PID: 885 (code=exited, status=0/SUCCESS)
     CPU: 24ms
```

It seems like the service is stuck in the "activating" state, and no matter how many times I restart, stop, disable, or re-enable the service, it remains in that state.
Here's my configuration file for reference:
I've intentionally commented out the sections related to customization and rebranding, but this didn't resolve the issue. Here's the relevant part of my config file:

```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "REDACTED",
    "WANonly": true,
    "_LANonly": true,
    "sessionKey": "REDACTED",
    "_port": 443,
    "_aliasPort": 443,
    "_redirPort": 80,
    "_redirAliasPort": 80,
    "trustedProxy": "REDACTED"
  },
  "domains": {
    "": {
      "certurl": "https://REDACTED.com",
      "title": "Remote Control Center",
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true,
      "_agentCustomization": {
        "_displayName": "Remote Agent",
        "_description": "Remote Agent provides seamless, secure, and efficient remote control and IT management solutions.",
        "_companyName": "REDACTED",
        "_serviceName": "Remote Agent",
        "_installText": "Text string to show in the agent installation dialog box",
        "_image": "remote.png",
        "_fileName": "remoteagent",
        "_foregroundColor": "#FFA500",
        "_backgroundColor": "#EE82EE"
      }
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "skipChallengeVerification": true,
    "production": false
  }
}
```
Any help or guidance would be greatly appreciated!
Thanks in advance for your time.

**Comment by si458** [comment]
you can run `journalctl -u meshcentral.service` and this will output logs
you can also check the `mesherrors.txt` file inside `meshcentral-data` for any errors WHY meshcentral crashed/not starting

**Reply by jbob007** [reply, marked as answer]
Thank you! I found it was a parsing error. I fixed it and it's working great again! For anyone having the same issue I copied the config.json content and used https://jsonlint.com to show me the location of the error. Hope this helps anyone else having the same issue!
